server/UserAuth/UserAuth.js

```diff
@@ -0,0 +1,74 @@
+require("../database/Database");
+const bcrypt = require('bcrypt');
+const UserModel = require('../database/models/User');
+const saltRounds = 10;
+
+function addUser(name, email, password) {
+    let saltGen;
+    return UserModel.findOne({email: email})
+    .then ( (res) => {
+        if (res){
+            console.log("Email already exists");
+            return 1;
+        } else {
+            return bcrypt.genSalt(saltRounds)
```

Comment:
We can simplify this part by taking full advantage of the hash method:

As per documentation:
```
hash(data, salt, cb)
data - [REQUIRED] - the data to be encrypted.
salt - [REQUIRED] - the salt to be used to hash the password. if specified as a number then a salt will be generated with the specified number of rounds and used (see example under Usage).
cb - [OPTIONAL] - a callback to be fired once the data has been encrypted. uses eio making it asynchronous. If cb is not specified, a Promise is returned if Promise support is available.
err - First parameter to the callback detailing any errors.
encrypted - Second parameter to the callback providing the encrypted form.
```
So we can specify the number of rounds in the hash method instead of specifying a salt. I think we should also promisify this

Reply:
I am not using the hash method because I am currently storing the salt. The .hash() method will only return the resulting password, and not the salt.